lib/travis/build/config.rb

```diff
@@ -33,15 +53,30 @@ def go_version_aliases_hash
           trusty: ENV.fetch('TRAVIS_BUILD_APT_SOURCE_WHITELIST_TRUSTY', '')
         },
         apt_whitelist_skip: ENV.fetch('TRAVIS_BUILD_APT_WHITELIST_SKIP', ''),
+        cabal_default: ENV.fetch('TRAVIS_BUILD_CABAL_DEFAULT', '1.22'),
         enable_debug_tools: ENV.fetch(
           'TRAVIS_BUILD_ENABLE_DEBUG_TOOLS',
           ENV.fetch('TRAVIS_ENABLE_DEBUG_TOOLS', '')
         ),
         etc_hosts_pinning: ENV.fetch(
           'TRAVIS_BUILD_ETC_HOSTS_PINNING', ENV.fetch('ETC_HOSTS_PINNING', '')
         ),
-        ghc_default: ENV.fetch('TRAVIS_BUILD_GHC_DEFAULT', '7.8.4'),
-        cabal_default: ENV.fetch('TRAVIS_BUILD_CABAL_DEFAULT', '1.18'),
+        ghc_default: ENV.fetch('TRAVIS_BUILD_GHC_DEFAULT', '7.10.3'),
+        ghc_version_aliases: ENV.fetch(
+          'TRAVIS_BUILD_GHC_VERSION_ALIASES', (
+            latest_semver_aliases(
+              '6' => '6.12.3',
+              '7' => '7.10.3',
+              '8' => '8.0.2'
+            ).merge(
+              '7.8.x' => '7.8.4',
```

Comment:
I assume this is about the values you canuse in https://docs.travis-ci.com/user/languages/haskell#Specifying-the-GHC-version ? If so, this looks good to me.

As for the rest of the script, I'm still digesting the code... :-)